src/IssueViz.Security.UnitTests/Taint/TaintList/TaintIssuesControlViewModelTests.cs

```diff
@@ -701,6 +716,34 @@ public void AnalysisInformation_IssuesChanged_RaisesPropertyChanged()
             testSubject.AnalysisInformation.Should().BeSameAs(analysisInformation);
         }
 
+        [TestMethod]
+        [DataRow(null, ServerType.SonarCloud, nameof(ServerType.SonarCloud), true)]
+        [DataRow(null, null, "", false)]
+        [DataRow(ServerType.SonarCloud, null, "", true)]
+        [DataRow(ServerType.SonarCloud, ServerType.SonarCloud, nameof(ServerType.SonarCloud), false)]
+        [DataRow(ServerType.SonarCloud, ServerType.SonarQube, nameof(ServerType.SonarQube), true)]
+        public void ActiveDocChanged_ExpectedServerTypeSetOnlyWhenItIsChanged(ServerType? originalServerType,
+            ServerType? newServerType, string expectedValue, bool expectedRaiseEvent)
+        {
+            var activeDocumentTrackerMock = new Mock<IActiveDocumentTracker>();
+            var sonarQubeServiceMock = new Mock<ISonarQubeService>();
+            SetServerType(originalServerType, sonarQubeServiceMock);
+            var testSubject = CreateTestSubject(activeDocumentTracker: activeDocumentTrackerMock.Object, sonarQubeService:sonarQubeServiceMock.Object);
```

Comment:
Cosmetics: new line before testSubject, to separate between the arrange and the act

Reply:
the test subject creation here is in the arrange

Reply:
It's messy to read imo. TestSubject is always declared separately to ease on the reader. It's one line, change it :D

Reply:
Yeah but I already have RaiseActiveDocumentChangedEvent(activeDocumentTrackerMock); separated by an empty line, adding another one would be more confusing, because it would make 4 paragraphs. The creation of test subject is not really important here to warrant an extra line, IMO. Alternatively, this is what I used to do before, you could separate the AAA by 2 empty lines and then add single empty lines in the As if the visibility becomes a problem